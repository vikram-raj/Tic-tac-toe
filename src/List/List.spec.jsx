import React from 'react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import List from './List';

configure({adapter: new Adapter()});

describe('ToDo', () => {
  it('renders', () => {
    const list = shallow(<List />);

    expect(list).toBeTruthy();
  });

  it('renders a list of to do based on state.toDos', () => {
    const list = shallow(<List />);
    list.setState({
      toDos: [
        { complete: false, text: 'One' },
        { complete: true, text: 'Two' }
      ]
    });

    expect(toJson(list)).toMatchSnapshot();
  });
});