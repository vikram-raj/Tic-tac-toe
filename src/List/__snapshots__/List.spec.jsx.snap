// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo renders a list of to do based on state.toDos 1`] = `
<div
  className="list"
>
  <h1
    className="list__header"
  >
    todos
  </h1>
  <StateToDo
    submit={[Function]}
  />
  <ToDo
    complete={false}
    deleteToDo={[Function]}
    key="One"
    text="One"
    toggleCompletion={[Function]}
  />
  <ToDo
    complete={true}
    deleteToDo={[Function]}
    key="Two"
    text="Two"
    toggleCompletion={[Function]}
  />
</div>
`;
